### DISCUSSION TITLE
Exposing the Datagrid commands.

### DISCUSSION THREAD

**Discussion by gallwood** [Megabit/Blazorise#1912] [Q&A]
Hi,

I'm using the DataGrid extension. It's very functional and does almost everything I need. There is one thing I wanted though that I can't seem to get it to do.

I have allowed the grid to be editable, but the New row command, I wanted to be able to trigger this command from outside of the DataGrid, from a seperate Button on my page. However, the OnNewCommand is protected. I guess this would be true for the other Commands, well, Edit and Delete anyway.

Any suggestion would be welcome.

Thanks

Graham

**Comment by stsrki** [comment]
There was some talking about exposing some of the APIs but nothing is yet done. Soon anything that can be done is for 0.9.4.

@David-Moreira it might be a good idea to see how this can be implemented?

**Comment by David-Moreira** [comment]
Yea for now the closest to "outside of the data grid" is the new Button Row on 0.9.3 which is instead located on the pager, where you can place your own buttons and trigger the new edit and delete commands which are exposed there.

But we'll definitely have to think about a good way to expose them more globally to give more liberty to the developer. I'll think about it and let you know when I have a concrete ideia.

**Comment by David-Moreira** [comment]
I was thinking about this even further... maybe if you really really really wanted. 
And only as a workaround you could have a very dirty solution. You could create the command button but have it be hidden and give it an id. 

And then trigger the button from anywhere using javascript (with interop if needed).

**Reply by gallwood** [reply]
Ooh. That’s a sneaky idea. Thanks

**Reply by David-Moreira** [reply]
@gallwood and I think you could also do it through c# actually, take a quick look at #1914
I posted a brief, yet untested example.

**Reply by gallwood** [reply]
Thanks @David-Moreira , that is an elegant work around.
